data/types.go

```diff
@@ -284,7 +285,12 @@ func (d *DelegatedRole) MarshalJSON() ([]byte, error) {
 func (d *DelegatedRole) UnmarshalJSON(b []byte) error {
 	type delegatedRoleAlias DelegatedRole
 
-	if err := json.Unmarshal(b, (*delegatedRoleAlias)(d)); err != nil {
+	// Prepare decoder
+	dec := json.NewDecoder(bytes.NewReader(b))
+	dec.DisallowUnknownFields()
```

Comment:
I'm a little worried about disallowing unknown fields here. TUF spec should allow unknown fields and keep them when computing hashes.

https://github.com/theupdateframework/specification/blob/master/tuf-spec.md#document-formats--document-formats

@mnm678 or other TUF spec compliance folks WDYT?

Reply:
This is a good point. I noticed that `keyid_hash_algorithms` was in go-tuf's keys, but not in the spec. I failed to find the part of the spec you linked which says this is allowable.

Reply:
This is interesting and a bit confusing. What is the source of truth? Specification or implementations?

Reply:
`keyid_hash_algorithms` was something that was added to python-tuf (the reference implementation) on an ad-hoc basis IIRC. It became something other implementations had to consider in order to interoperate. I'd say it should be in the pre-1.0 TUF spec, but obviously not in 1.0 or after. What do others think?

And yes, unknown fields should be ignored. Didn't we clarify this in the spec?

Reply:
The spec has long suggested supporting unknown fields for backwards compatibility, we further clarified this last year ([discussion](https://github.com/theupdateframework/specification/issues/163), [PR](https://github.com/theupdateframework/specification/pull/169)) to state that implementations must include unrecognised fields when calculating hashes and signatures and that implementations should preserve the data when re-serialising.